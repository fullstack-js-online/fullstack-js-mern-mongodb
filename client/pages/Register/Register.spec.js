import React from 'react'
import { mount } from 'enzyme'

import { RegisterPage } from './index'

describe('Register page', () => {
  const props = {
    dispatch: jest.fn()
  }

  it('should match snapshot', () => {
    expect(mount(<RegisterPage {...props} />)).toMatchSnapshot()
  })
})
