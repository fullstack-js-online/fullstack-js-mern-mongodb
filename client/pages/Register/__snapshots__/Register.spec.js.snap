// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page should match snapshot 1`] = `
<RegisterPage
  dispatch={[MockFunction]}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      title="Register"
      titleAttributes={Object {}}
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        title="Register"
        titleAttributes={Object {}}
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
  <RegisterForm
    dispatch={[MockFunction]}
    initialValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "email",
          "name",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "name": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <div
      className="container mx-auto flex justify-center"
    >
      <div
        className="max-w-md-1/2 mx-3 my-12 w-full"
      >
        <h1
          className="text-center font-primary font-semibold text-gold"
        >
          Register
        </h1>
        <div
          className="bg-white shadow-md my-12 p-16 rounded w-full"
        >
          <Formik
            enableReinitialize={false}
            initialValues={
              Object {
                "email": "",
                "name": "",
                "password": "",
              }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": undefined,
                "_nodes": Array [
                  "password",
                  "email",
                  "name",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "email": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": true,
                      "undefined": false,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "name": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "password": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "max": true,
                      "min": true,
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          >
            <form
              onSubmit={[Function]}
            >
              <Textbox
                data-testid="name-field"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your name"
                type="name"
                value=""
              >
                <div
                  className="mb-5"
                >
                  <input
                    className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter your name"
                    type="name"
                    value=""
                  />
                </div>
              </Textbox>
              <Textbox
                data-testid="email-field"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your email"
                type="email"
                value=""
              >
                <div
                  className="mb-5"
                >
                  <input
                    className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter your email"
                    type="email"
                    value=""
                  />
                </div>
              </Textbox>
              <Textbox
                data-testid="password-field"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your password"
                type="password"
                value=""
              >
                <div
                  className="mb-5"
                >
                  <input
                    className="focus:outline-none w-full p-6 bg-brown-lightest text-brown"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter your password"
                    type="password"
                    value=""
                  />
                </div>
              </Textbox>
              <Button
                data-testid="submit-button"
                loading={false}
                type="submit"
              >
                <button
                  className="focus:outline-none rounded-sm hover:bg-emerald-light px-4 py-5 w-full text-white text-xl bg-emerald"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </button>
              </Button>
            </form>
          </Formik>
        </div>
      </div>
    </div>
  </RegisterForm>
</RegisterPage>
`;
